### ISSUE TITLE
XDM is not capturing any downloads in Firefox

### ISSUE THREAD

**Issue by C0deSeeker** [subhra74/xdm#235]
XDM is not capturing any downloads in Firefox. Also, while playing YouTube videos, there's no pop-up for video downloading.
XDM download works perfectly if I manually add links. 

**To Reproduce**
1. Go to any website with downloadable content.
2. Click on the download link
4. Nothing happens.

**Expected behavior**
XDM should capture downloads when a downloadable link is clicked in the browser.

 - OS: Windows 10
 - Browser: Firefox 77.0.1
 - XDM addon Version: 2.1
 - XDM Version: 7.2.11

**Generated log using below method**
 
![image](https://user-images.githubusercontent.com/36724358/85581445-37574700-b65a-11ea-9484-21d8e433655f.png)

**Additional context**

**Comment by arminolat** [comment]
I can confirm that the same thing happens to me since yesterday.

**Comment by ogasule** [comment]
Same here and its not only unique to firefox, other browsers too are affected by this.

**Comment by subhra74** [comment]
Are you using any VPN/Proxy or some security configuration related to Firefox?

**Comment by ngbede** [comment]
No I am not. All this issues emerged after the previous update.

**Comment by C0deSeeker** [comment]
@subhra74 Yes, you seem to be right. Although, I wasn't using any VPN or proxy, I had changed numerous about:config preferences in Firefox. Guessing this to be the potential cause, I made a new profile in the browser. And bravo! Everything is working fine as it should be.

**Comment by nadeem49** [comment]
Go to `about:config` in firefox and change `privacy.firstparty.isolate `= true to **false**. I had the same issue. when i changed this, it was capturing my downloads fine.

**Comment by Waterfoxer** [comment]
> 
> 
> Go to `about:config` in firefox and change `privacy.firstparty.isolate `= true to **false**. I had the same issue. when i changed this, it was capturing my downloads fine.

You are a lifesaver. Though I like that set to TRUE for security, it's indeed what caused the problem.  Bravo.

**Comment by TheHighGroundWins** [comment]
I'm using librewolf and still having this issue. package is installed as system package.
it works fine on firefox, but on librewolf the extension simply won't become blue. 
` privacy.firstparty.isolate` is set to  false.
For now I'll just use the download manager manually.
I tried disabling many security options to no avail.

**Comment by C0deSeeker** [comment]
> I'm using librewolf and still having this issue. package is installed as system package. it works fine on firefox, but on librewolf the extension simply won't become blue. ` privacy.firstparty.isolate` is set to false. For now I'll just use the download manager manually. I tried disabling many security options to no avail.

I also faced this same issue when I migrated to LibreWolf. You need to go to the LibreWolf settings, and in the 'General' section uncheck 'Always ask you where to save file' option for downloads. XDM should work as expected. You can also search for this option in LibreWolf settings.

**Comment by TheHighGroundWins** [comment]
> > I'm using librewolf and still having this issue. package is installed as system package. it works fine on firefox, but on librewolf the extension simply won't become blue. ` privacy.firstparty.isolate` is set to false. For now I'll just use the download manager manually. I tried disabling many security options to no avail.
> 
> I also faced this same issue when I migrated to LibreWolf. You need to go to the LibreWolf settings, and in the 'General' section uncheck 'Always ask you where to save file' option for downloads. XDM should work as expected. You can also search for this option in LibreWolf settings.

From the fact that the extension icon is still greyed out while its blue in firefox. Seems to suggest something off on my end.
